test/DebugInfo/DebugUnstructuredControlFlow.cl

```diff
@@ -0,0 +1,29 @@
+// RUN: %clang_cc1 -triple spir64-unknown-unknown -cl-std=CL2.0 -O0 -debug-info-kind=standalone -emit-llvm %s -o %t.ll
+// RUN: llvm-as %t.ll -o %t.bc
+// RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_unstructured_loop_controls -spirv-text -o %t.spt
+// RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+// RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_unstructured_loop_controls -o %t.spv
+// RUN: llvm-spirv -r %t.spv -o %t.bc
+// RUN: llvm-dis < %t.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+// Test that no debug info instruction is inserted between LoopControlINTEL and
+// Branch instructions. Otherwise, debug info interferes with SPIRVToLLVM
+// translation of structured flow control.
+//
+// Currently, Line DebugInfo instructions are still present between
+// LoopControlINTEL and Branch instructions. This does not affect SPIRVToLLVM
+// translation, however should be fixed separately.
+
+kernel
+void sample() {
+  #pragma clang loop unroll(full)
+  for(;;);
+}
+
+// CHECK-SPIRV: 2 LoopControlINTEL 1
+// CHECK-SPIRV-NOT: ExtInst
+// CHECK-SPIRV: {{[0-9]+}} Line {{[0-9]+}} {{[0-9]+}} {{[0-9]+}}
```

Comment:
To clarify: is it guaranteed that OpLine will land here? If there's no certainty as to its placement, we could simply omit the check.

Reply:
I think it should present before Branch instruction.